### DISCUSSION TITLE
Would be a good feature to return stack allocated bytes by pinning it, then moving away from it, then return a Span<T>, instead of having to copy it to heap first since stack will be destoryed

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#62197] [Ideas]
Recently I've written an algorithm that does so much math I wanted to just returned the bytes I've allocated in the stack for performance reason but I can't find a way to do it without copying to heap, or returning by values.
The stack allocated memory will change, I know I can probably hack it by returning a struct that has 256 single byte variable and a length that tells how much of them were the values actually returned, I could probably use a shared memory passed in from the parameter, but also would be nice if I can get a pointer to this memory allocated in the stack and make sure it doesn't get destroyed.

**Comment by Xyncgas** [comment]
Which is possible already if you inline the function being called to the function calling it, now you have the same stack, so it can be preserved

**Comment by gfoidl** [comment]
It's still the same as in https://github.com/dotnet/runtime/issues/62127 (and the same answers) for

> wanted to just returned the bytes I've allocated in the stack for performance reason but I can't find a way to do it without copying to heap, or returning by values.
